File: demo/src/main/java/com/bytedance/scenedemo/extreme_case/Case2Scene.java
package com.bytedance.scenedemo.extreme_case;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.bytedance.scene.group.GroupScene;
import com.bytedance.scene.interfaces.PopOptions;
import com.bytedance.scene.interfaces.PushOptions;
import com.bytedance.scenedemo.R;
import com.bytedance.scenedemo.lifecycle.EmptyScene;
import com.bytedance.scenedemo.utility.ColorUtil;

/**
 * Created by JiangQi on 9/5/18.
 */
public class Case2Scene extends GroupScene {

    @NonNull
    @Override
    public ViewGroup onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return (ViewGroup) inflater.inflate(R.layout.basic_layout, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        getView().setBackgroundColor(ColorUtil.getMaterialColor(getResources(), 0));

        TextView name = getView().findViewById(R.id.name);
        name.setVisibility(View.GONE);

        Button btn = getView().findViewById(R.id.btn);
        btn.setText(R.string.case_pop_many_btn_1);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getNavigationScene().push(EmptyScene.class, null, new PushOptions.Builder().build());
                for (int i = 0; i < 100; i++) {
                    getNavigationScene().pop();
                }
            }
        });

        Button btn2 = getView().findViewById(R.id.btn2);
        btn2.setVisibility(View.VISIBLE);
        btn2.setText(R.string.case_pop_many_btn_2);
        btn2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getNavigationScene().push(EmptyScene.class, null, new PushOptions.Builder().build());
                getNavigationScene().pop(new PopOptions.Builder().setPopUtilPredicate(new PopOptions.CountUtilPredicate(100)).build());
            }
        });

        Button btn3 = getView().findViewById(R.id.btn3);
        btn3.setVisibility(View.VISIBLE);
        btn3.setText(R.string.case_pop_many_btn_3);
        btn3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(getActivity(), R.string.case_pop_many_toast_1, Toast.LENGTH_SHORT).show();
                new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        for (int i = 0; i < 100; i++) {
                            getNavigationScene().pop();
                        }
                        getNavigationScene().push(EmptyScene.class, null, new PushOptions.Builder().build());
                        Toast.makeText(getActivity(), R.string.case_pop_many_toast_2, Toast.LENGTH_SHORT).show();
                    }
                }, 3000);
            }
        });
    }

}

File: demo/src/main/java/com/bytedance/scenedemo/extreme_case/Case2Scene.kt
package com.bytedance.scenedemo.extreme_case

import com.bytedance.scene.group.GroupScene
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.os.Handler
import com.bytedance.scenedemo.R
import android.widget.TextView
import com.bytedance.scene.interfaces.PushOptions
import com.bytedance.scene.interfaces.PopOptions
import com.bytedance.scene.interfaces.PopOptions.CountUtilPredicate
import android.widget.Toast
import android.os.Looper
import android.view.View
import android.widget.Button
import com.bytedance.scenedemo.lifecycle.EmptyScene
import com.bytedance.scenedemo.utility.ColorUtil

/**
 * Created by JiangQi on 9/5/18.
 */
class Case2Scene : GroupScene() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle?): ViewGroup {
        return inflater.inflate(R.layout.basic_layout, container, false) as ViewGroup
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        view.setBackgroundColor(ColorUtil.getMaterialColor(resources, 0))
        val name = view.findViewById<TextView>(R.id.name)
        name.visibility = View.GONE
        val btn = view.findViewById<Button>(R.id.btn)
        btn.setText(R.string.case_pop_many_btn_1)
        btn.setOnClickListener {
            navigationScene!!.push(EmptyScene::class.java, null, PushOptions.Builder().build())
            for (i in 0..99) {
                navigationScene!!.pop()
            }
        }
        val btn2 = view.findViewById<Button>(R.id.btn2)
        btn2.visibility = View.VISIBLE
        btn2.setText(R.string.case_pop_many_btn_2)
        btn2.setOnClickListener {
            navigationScene!!.push(EmptyScene::class.java, null, PushOptions.Builder().build())
            navigationScene!!.pop(PopOptions.Builder().setPopUtilPredicate(CountUtilPredicate(100)).build())
        }
        val btn3 = view.findViewById<Button>(R.id.btn3)
        btn3.visibility = View.VISIBLE
        btn3.setText(R.string.case_pop_many_btn_3)
        btn3.setOnClickListener {
            Toast.makeText(activity, R.string.case_pop_many_toast_1, Toast.LENGTH_SHORT).show()
            Handler(Looper.getMainLooper()).postDelayed({
                for (i in 0..99) {
                    navigationScene!!.pop()
                }
                navigationScene!!.push(EmptyScene::class.java, null, PushOptions.Builder().build())
                Toast.makeText(activity, R.string.case_pop_many_toast_2, Toast.LENGTH_SHORT).show()
            }, 3000)
        }
    }
}